modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java

```diff
@@ -0,0 +1,408 @@
+		try {
+			assertNull(loginLogoutClient
+					           .login(loggedInUsername, oldPassword, asServer.getInstance().getHosts().get("default")),
+			           "Couldn't login with new password");
+		} catch (Exception e) {
+			ex = e;
+		}
+		assertNotNull(ex, "User able to login to system using old password");
+		assertEquals(ex.getMessage(), "Error while login as " + loggedInUsername,
+		             "User able to login to system using old password");
+
+		//Revert back to old password
+		userManagementClient.changePasswordByUser(newPassword, oldPassword);
+		loginLogoutClient = new LoginLogoutClient(asServer);
+		assertNotNull(loginLogoutClient
+				              .login(loggedInUsername, oldPassword, asServer.getInstance().getHosts().get("default")),
+		              "Failed to revert old password");
+	}
+
+	@Test(groups = "wso2.as", description = "Update role name", dependsOnMethods = { "testAddRoleToUser" })
+	public void testUpdateRoleName() throws Exception {
+		String newRoleName = "new_" + roleName;
+		userManagementClient.updateRoleName(roleName, newRoleName);
+		assertTrue(userManagementClient.roleNameExists(newRoleName), "Role rename failed");
+
+		FlaggedName[] rolesOfUser = userManagementClient.getRolesOfUser(username, "*", 10);
+		for (FlaggedName flaggedName : rolesOfUser) {
+			if (roleName.equals(flaggedName.getItemName())) {
+				assertFalse(flaggedName.getSelected(), "User still has previous role name");
+			} else if (newRoleName.equals(flaggedName.getItemName())) {
+				assertTrue(flaggedName.getSelected(), "User don't have renamed role");
+			}
+		}
+		roleName = newRoleName;
+	}
+
+	@Test(groups = "wso2.as", description = "Change password by username", dependsOnMethods = {
+			"testLoginWithNewUser" })
+	public void testChangePasswordByUsername() throws Exception {
+		String newPassword = "new_" + password;
+		userManagementClient.changePassword(username, newPassword);
+		String oldUsername = username;
+		if (userMode == TestUserMode.TENANT_ADMIN) {
+			username += "@" + asServer.getContextTenant().getDomain();
+		}
+		String newSessionCookie =
+				loginLogoutClient.login(username, newPassword, asServer.getInstance().getHosts().get("default"));
+		assertNotNull(newSessionCookie, "Can't login after change password");
+
+		//try to login with old password
+		Exception ex = null;
+		try {
+			assertNull(loginLogoutClient.login(username, password, asServer.getInstance().getHosts().get("default")),
+			           "Couldn't login with new password");
+		} catch (Exception e) {
+			ex = e;
+		}
+		assertNotNull(ex, "User able to login to system using old password");
+		assertEquals(ex.getMessage(), "Error while login as " + username,
+		             "User able to login to system using old password");
+
+		if (userMode == TestUserMode.TENANT_ADMIN) {
+			username = oldUsername;
+		}
+		password = newPassword;
+	}
+
+	@Test(groups = "wso2.as", description = "Deleting a user", dependsOnMethods = { "testAddRemoveUsersOfRole" })
```

Comment:
If a method in a test class as a depends on then as a practice need to define depends on for all other methods.